packages/md-enhance/src/node/markdown-it/playground/uno.ts

```diff
@@ -0,0 +1,66 @@
+import { endsWith, keys } from "vuepress-shared/node";
+
+import { compressToEncodedURIComponent as encode } from "./ventors/lzstring.js";
+import type {
+  PlaygroundData,
+  PlaygroundOptions,
+  UnoPresetPlaygroundOptions,
+} from "../../typings/index.js";
+import { logger } from "../../utils.js";
+
+/** Gets a query string representation (hash + queries) */
+export const generateUnoURL = (
+  service: string,
+  inputHTML: string,
+  customCSS: string,
+  customConfigRaw: string,
+): string => {
+  const getParam = (key: string, value: string, sign: string = "&"): string => {
+    if (value) return `${sign}${key}=${encode(value)}`;
+
+    return "";
+  };
```

Comment:
I think the value is not using any scope values, so these functions should be moved out of the function for perf consideration, as we do not have to create new ones in each call.

Reply:
Make sense, I have learned from your comment. thx.